src/Appwrite/Migration/Version/V19.php

```diff
@@ -0,0 +1,180 @@
+                $this->alterPermissionIndex($collectionTable);
+            }
+        }
+    }
+
+    /**
+     * Migrate all Collections.
+     *
+     * @return void
+     */
+    private function migrateCollections(): void
+    {
+        foreach ($this->collections as $collection) {
+            $id = $collection['$id'];
+
+            Console::log("Migrating Collection \"{$id}\"");
+
+            if (!in_array($id, ['files', 'collections'])) {
+                $this->alterPermissionIndex($id);
+            }
+
+            usleep(50000);
+        }
+    }
+
+    /**
+     * Fix run on each document
+     *
+     * @param Document $document
+     * @return Document
+     */
+    protected function fixDocument(Document $document): Document
+    {
+        switch ($document->getCollection()) {
+            case 'projects':
+                /**
+                 * Bump version number.
+                 */
+                $document->setAttribute('version', '1.3.0');
+
+                /**
+                 * Set default passwordHistory
+                 */
+                $document->setAttribute('auths', array_merge([
+                    'passwordHistory' => 0,
+                    'passwordDictionary' => false,
+                ], $document->getAttribute('auths', [])));
+                break;
+            case 'users':
+                /**
+                 * Default Password history
+                 */
+                $document->setAttribute('passwordHistory', $document->getAttribute('passwordHistory', []));
+                break;
+            case 'teams':
+                /**
+                 * Default prefs
+                 */
+                $document->setAttribute('prefs', $document->getAttribute('prefs', new \stdClass()));
+                break;
+            case 'attributes':
+                /**
+                 * Default options
+                 */
+                $document->setAttribute('options', $document->getAttribute('options', new \stdClass()));
+                break;
+            case 'buckets':
+                /**
+                 * Set the bucket permission in the metadata table
+                 */
+                try {
+                    $internalBucketId = "bucket_{$this->project->getInternalId()}";
+                    $permissions = $document->getPermissions();
+                    $fileSecurity = $document->getAttribute('fileSecurity', false);
+                    $this->projectDB->updateCollection($internalBucketId, $permissions, $fileSecurity);
+                } catch (\Throwable $th) {
+                    Console::warning($th->getMessage());
+                }
+                break;
+        }
+
+        return $document;
+    }
+
+    protected function alterPermissionIndex($collectionName): void
+    {
+        $collectionName = "`{$this->projectDB->getDefaultDatabase()}`.`_{$this->project->getInternalId()}_{$collectionName}_perms`";
```

Comment:
Do we need this or can we use the parameter?

Reply:
Now that we are doing the alter in a single transaction I can put it inline.
Done!